src/Microsoft.AspNet.Server.Kestrel/KestrelServerInformation.cs

```diff
@@ -107,5 +117,59 @@ private static bool GetNoDelay(IConfiguration configuration)
 
             return true;
         }
+
+        private static TimeSpan GetKeepAliveTimeout(IConfiguration configuration)
+        {
+            var keepAliveTimeoutString = configuration["server.keepAliveTimeout"];
+
+            if (string.IsNullOrEmpty(keepAliveTimeoutString))
+            {
+                return TimeSpan.FromSeconds(160);
+            }
+
+            TimeSpan keepAliveTimeout;
+            if (TimeSpan.TryParse(keepAliveTimeoutString, out keepAliveTimeout))
+            {
+                return keepAliveTimeout;
+            }
+
+            return TimeSpan.FromSeconds(160);
+        }
+
+        private static TimeSpan GetHeadersCompleteTimeout(IConfiguration configuration)
+        {
+            var headersCompleteTimeoutString = configuration["kestrel.headersCompleteTimeout"];
+
+            if (string.IsNullOrEmpty(headersCompleteTimeoutString))
+            {
+                return TimeSpan.FromSeconds(60);
+            }
+
+            TimeSpan headersCompleteTimeout;
+            if (TimeSpan.TryParse(headersCompleteTimeoutString, out headersCompleteTimeout))
+            {
+                return headersCompleteTimeout;
+            }
+
+            return TimeSpan.FromSeconds(60);
+        }
+
+        private static TimeSpan GetBodyKeepAliveTimeout(IConfiguration configuration)
+        {
+            var bodyKeepAliveTimeoutString = configuration["kestrel.bodyKeepAliveTimeout"];
+
+            if (string.IsNullOrEmpty(bodyKeepAliveTimeoutString))
+            {
+                return TimeSpan.FromSeconds(60);
+            }
+
+            TimeSpan keepAliveTimeout;
+            if (TimeSpan.TryParse(bodyKeepAliveTimeoutString, out keepAliveTimeout))
+            {
+                return keepAliveTimeout;
+            }
+
+            return TimeSpan.FromSeconds(60);
+        }
```

Comment:
The three methods above are extremely similar to each other. Refactor into a single method with parameters for each timeout?